RoboJackets/robocup-software
# Nmark tactic

## Description
This implements the nmark tactic, which marks the n enemies closest to the ball with the robots of ours that are closest to them. For now, ignore the roughness of the `defensive_clear.py` file, it will be uncommented and filled in once pivot_kick from #1644  is merged in.

## Associated Issue
https://app.clickup.com/t/y3fvx9

## Steps to test
### Test Case 1
1. Start sim
2. Drag enemy robots (blue) around the field
3. Kick ball around

Expected result: Our 2 closest robots to the 2 enemies closest to the ball should move to mark those enemies, regardless of where the ball is. (I know this wording is a little unclear but the behavior makes sense when you see it.)

### Test Case 2
1. Change the number in the `init()` of `play/defensive_clear.py` that controls how many robots nmark handles.
2. Repeat Test Case 1.

Expected result: NMark works the same, but with more (or less) than 2 robots.
